app/models/group_offer.rb

```diff
@@ -25,6 +25,31 @@ class GroupOffer < ApplicationRecord
 
   scope :in_department, (-> { where.not(department_id: nil) })
 
+  scope :active_group_assignments_between, lambda { |start_date, end_date|
+    joins(:group_assignments).merge(GroupAssignment.active_between(start_date, end_date))
+  }
+
+  def has_active_group_assignments_between?(start_date, end_date)
+    group_assignments.active_between(start_date, end_date).size.positive?
```

Comment:
Use `.exists?` instead of `.size.positive?`

Reply:
Your right, but how about `any?`